ui/src/components/VoxettaRecordButton.js

```diff
@@ -0,0 +1,150 @@
+        this.utteranceService = new UtteranceApiService();
+        this.audioStream; 
+    }
+  
+    /**
+     * If the user is not currently recording, begin recording using the Microphone 
+     * component. Otherwise, stop recording and save and display the just-recorded 
+     * audio file.
+     */
+    async recordHandler() {
+        if (!this.isRecording) {
+            try { 
+                await this.audioRecorder.initRecorder();
+            } catch(e) { 
+                console.log("Microphone blocked"); 
+            }
+            if (this.audioRecorder.startRecording()) {
+                this.isRecording = true;
+                this.styleOnButton(); 
+            }
+            this.audioStream = this.audioRecorder.stream;
+        } else {
+            this.isRecording = false;
+            this.styleOffButton();
+            this.handleFinish();
+            const audio = await this.audioRecorder.stopRecording();
+            if (audio.recordingUrl) {
+                // NOTE: removed for Monday presentation
+                //const audioSave = this.shadowRoot.getElementById("utterance");
+                //audioSave.src = audio.recordingUrl;            
+                //audioSave.style.display = "block";
+                this.utteranceService.saveAudio(audio);
+            }
+        }
+    }
+
+    /**
+     * Styles the recording button to its recording-state style.
+     */
+    styleOnButton() {
+        const button = this.shadowRoot.getElementById("record-button");
+        button.icon = "stop";
+        button.style.color = "white";
+        button.style.backgroundColor = "red";
+    }
+
+    /**
+     * Styles the recording button to its non-recording-state style.
+     */
+    styleOffButton() {
+        const button = this.shadowRoot.getElementById("record-button");
+        button.icon = "mic";
+        button.style.color = "#3c4043";
+        button.style.backgroundColor = "white";
+    }
+
+    /**
+     * Emits an event that causes the application to render a sound
+     * wave that corresponds to the current audio stream. 
+     */
+    handleStateChange() {
+        const event = new CustomEvent('record-state-change', {
+            detail: { 
+                isRecording: this.isRecording,
+                audioStream: this.audioStream
+            }
+        });
+        this.dispatchEvent(event);
+    }
+
+    /**
+     * Emits an event that causes a new prompt to be rendered
+     * on the recording page. 
+     */
+    handleFinish() {
+        const event = new CustomEvent('next-prompt', {});
+        this.dispatchEvent(event);
+    }
+    
+    render() {
+        this.handleStateChange();
+        return html`
+            <mwc-icon-button 
+                id="record-button"
+                icon="mic"
+                @click=${this.recordHandler}>
+            </mwc-icon-button>
+            <!-- <audio id="utterance" controls src="" style="display: none"></audio> -->
```

Comment:
Remove commented out code

Reply:
I was not sure if we were keeping this for the demo or for debugging purposes--will either remove or un-comment after I talk with Elvis & Eldrick